### ISSUE TITLE
macOS incorrectly renders sRGB colors in native display profile

### ISSUE THREAD

**Issue by AprilArcus** [microsoft/vscode#148863]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.67.0
- OS Version: macOS 12.3.1

Steps to Reproduce:

1. Load a theme designed in the sRGB color space
2. Sample the colors in Digital Color Meter
3. Colors have been scaled into the P3 color space, and appear excessively saturated.

I am using the Monokai Pro color theme. Note that it is tagged for the "sRGB" color space.

<img width="468" alt="image" src="https://user-images.githubusercontent.com/2045543/167048340-8abfad1c-d71d-470f-bb9a-6f75f41b8bf1.png">

The correct value for Monokai's red color is `#FF6188`, or `color(srgb 1 0.38 0.533)`)

Examine the color of a red-highlighted region of the screen using Digital Color Meter.

Note that in the sRGB space, the colors are clipped and appear visibly blown out:
<img width="449" alt="Screen Shot 2022-05-05 at 5 39 12 PM" src="https://user-images.githubusercontent.com/2045543/167048631-ea91b92a-cb46-4b78-aead-c972453e99fc.png">

In the native display profile space (Display P3 on this MacBookPro18,1), the colors are reported according to their input value:
<img width="440" alt="Screen Shot 2022-05-05 at 5 42 04 PM" src="https://user-images.githubusercontent.com/2045543/167049785-068b28d8-4cfd-408b-8a8d-5f3bcbb8d59f.png">

 — but these are sRGB values, and in the P3 space they should be scaled accordingly (to approximately `#ED6C89`, or `color(display-3 0.928 0.422 0.537)`).

**Comment by AprilArcus** [comment]
This is the [commit](https://github.com/textmate/textmate/commit/d70ccc7cdf4750eed00672cff77ddf23f4e96251) (dated Feb 18, 2013) that adds support for the `colorSpaceName: sRGB` key-value pair to the `.tmTheme` format. `sRGB` (with that exact capitalization) was the only valid key at the time of its introduction, and [remains so to this day](https://github.com/textmate/textmate/blob/7b51a4fbeb5e177717ce48cec09d51ba55b48ce1/Frameworks/theme/src/theme.cc#L247). 

I strongly feel that this key should be respected, and I also feel that sRGB is a more reasonable default choice for interpreting untagged hex triples, certainly compared to the device profile which could cause colors to vary drastically from machine to machine. CSS [specifies sRGB as the canonical color space for untagged hex triples](https://www.w3.org/TR/css-color-4/#untagged) on the Web, and [has done so since 1996](https://www.w3.org/TR/CSS1/#color-units).

It probably doesn't make sense to support other named color spaces in this field, since there is no specification to follow. Textmate's fallback, if sRGB is not explicitly named, is to use Apple's pre-2009 "Generic RGB Profile", with its notorious 1.8 gamma. I do not think VSCode should emulate this behavior. Support for advanced managed color features would be better accomplished through CSS4's [profile-tagged `color()`](https://www.w3.org/TR/css-color-4/#color-function) function and [device-independent `lab()`, `lch()`, `oklab()`, and `oklch()`](https://www.w3.org/TR/css-color-4/#lab-colors) functions, if and when Chromium gains support for them.

**Comment by aeschli** [comment]
We don't support the `colorSpaceName` property in our (VS Code theme) files. We expect all themes to define their colors in the sRGB colorspace. See [here](https://github.com/microsoft/vscode/issues/114591#issuecomment-764507645) for more information.
Isn't the problem that Chromium on the Mac uses the raw color profile as described in #87275?

There's a fix for this as described in https://github.com/microsoft/vscode/issues/87275#issuecomment-721986554.
Does that work for you?

**Comment by AprilArcus** [comment]
Unfortunately not. I did try that before opening this ticket, as you can see that it is among the "see also"s that I cited.

<img width="652" alt="Screen Shot 2022-05-09 at 1 44 03 PM" src="https://user-images.githubusercontent.com/2045543/167495115-a625f430-2638-4ed6-8b87-d5fc49db1de2.png">
<img width="660" alt="Screen Shot 2022-05-09 at 1 43 31 PM" src="https://user-images.githubusercontent.com/2045543/167495136-62d83ca1-372e-4080-aeec-ac18a38b89df.png">

**Comment by AprilArcus** [comment]
I observed that my startup flag `disable-color-correct-rendering` was set to `false` with this comment, which I found curious:

```
	// Enabled by default by VS Code to resolve color issues in the renderer
	// See https://github.com/microsoft/vscode/issues/51791 for details
	"disable-color-correct-rendering": true,
```

Then, I looked up the list of legal values for the "force-color-profile" flag: (see https://chromium.googlesource.com/chromium/src/+/master/chrome/browser/about_flags.cc#847), which are:

  * `srgb`
  * `display-p3-d65`
  * `color-spin-gamma24`
  * `scrgb-linear`
  * `hdr10`

I experimented with all combinations of `disable-color-rendering` (`true`, `false`, unset) and `force-color-profile` (`srgb`, `display-p3-d65`, unset) on my built-in display (MacBook Pro 16" 2021) with its default profile (`Apple XDR Display (P3-1600 nits)`), an sRGB profile (`Internet & Web (sRGB)`), and on my external display (Samsung Odyssey G9) using the ICC profile discovered via Plug-n-Play (`LC49G95T`). Here are the results:

<table>
  <caption>Built-in Display, Profile: Apple XDR Display (P3-1600 nits)</caption>
  <thead>
    <tr>
      <th scope="col"></th>
      <th scope="col"><code>"force-color-profile": "srgb"</code></th>
      <th scope="col"><code>"force-color-profile": "display-p3-d65"</code></th>
      <th scope="col"><code>"force-color-profile":</code> unset</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th scope="row"><code>"disable-color-correct-rendering": false</code></th>
      <td><b>correct</b></td>
      <td><b>correct</b></td>
      <td>oversaturated</td>
    </tr>
    <tr>
      <th scope="row"><code>"disable-color-correct-rendering": true</code></th>
      <td>oversaturated</td>
      <td>oversaturated</td>
      <td>oversaturated</td>
    </tr>
    <tr>
      <th scope="row"><code>"disable-color-correct-rendering":</code> unset</th>
      <td>oversaturated</td>
      <td><b>correct</b></td>
      <td>oversaturated</td>
    </tr>
  </tbody>
</table>

<table>
  <caption>Built-in Display, Profile: Internet & Web (sRGB)</caption>
  <thead>
    <tr>
      <th scope="col"></th>
      <th scope="col"><code>"force-color-profile": "srgb"</code></th>
      <th scope="col"><code>"force-color-profile": "display-p3-d65"</code></th>
      <th scope="col"><code>"force-color-profile":</code> unset</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th scope="row"><code>"disable-color-correct-rendering": false</code></th>
      <td><b>correct</b></td>
      <td><b>correct</b></td>
      <td><b>correct</b></td>
    </tr>
    <tr>
      <th scope="row"><code>"disable-color-correct-rendering": true</code></th>
      <td><b>correct</b></td>
      <td><b>correct</b></td>
      <td><b>correct</b></td>
    </tr>
    <tr>
      <th scope="row"><code>"disable-color-correct-rendering":</code> unset</th>
      <td><b>correct</b></td>
      <td><b>correct</b></td>
      <td><b>correct</b></td>
    </tr>
  </tbody>
</table>


<table>
  <caption>External Display, Profile: LC49G95T</caption>
  <thead>
    <tr>
      <th scope="col"></th>
      <th scope="col"><code>"force-color-profile": "srgb"</code></th>
      <th scope="col"><code>"force-color-profile": "display-p3-d65"</code></th>
      <th scope="col"><code>"force-color-profile":</code> unset</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th scope="row"><code>"disable-color-correct-rendering": false</code></th>
      <td><b>correct</b></td>
      <td><b>correct</b><br />(rounding errors)</td>
      <td>oversaturated</td>
    </tr>
    <tr>
      <th scope="row"><code>"disable-color-correct-rendering": true</code></th>
      <td>oversaturated</td>
      <td>oversaturated</td>
      <td>oversaturated</td>
    </tr>
    <tr>
      <th scope="row"><code>"disable-color-correct-rendering":</code> unset</th>
      <td><b>correct</b></td>
      <td><b>correct</b><br />(rounding errors)</td>
      <td>oversaturated</td>
    </tr>
  </tbody>
</table>

**Comment by AprilArcus** [comment]
Here is where color-correct rendering is disabled in the default config for new installations.

https://github.com/microsoft/vscode/blob/32d40cf44e893e87ac33ac4f08de1e5f7fe077fc/src/main.js#L284-L286

**Comment by AprilArcus** [comment]
Any news on this issue?

**Comment by AprilArcus** [comment]
This appears to be fixed on the current version of VS Code!

**Comment by canpoyrazoglu** [comment]
The latest version broke P3 color space display for me and everything started to look washed out again (this was fixed about a year ago).

**Comment by ZaquL** [comment]
Bisect someone?

**Comment by deepak1556** [comment]
@AprilArcus your issue should be addressed since 1.70, for reference https://github.com/microsoft/vscode/issues/152816#issuecomment-1183033252 

P3 display issue is being tracked under https://github.com/microsoft/vscode/issues/161028
